osu.Game/Overlays/Mods/ModSection.cs

```diff
@@ -159,12 +159,16 @@ public void DeselectAll()
         /// </summary>
         /// <param name="modTypes">The types of <see cref="Mod"/>s which should be deselected.</param>
         /// <param name="immediate">Whether the deselection should happen immediately. Should only be used when required to ensure correct selection flow.</param>
-        public void DeselectTypes(IEnumerable<Type> modTypes, bool immediate = false)
+        /// <param name="newSelection">If this deselection is triggered by a user selection, this should contain the newly selected type. This type will never be deselected, even if it matches one provided in <paramref name="modTypes"/>.</param>
+        public void DeselectTypes(IEnumerable<Type> modTypes, bool immediate = false, Mod newSelection = null)
```

Comment:
Not too happy about this new param existing,  but `DeselectTypes()` needs to disappear as a method at some point as it's doing a very crude approximation of the mod incompatibility logic anyways, so I'm willing to look away here.